pkg/stargate/deployments.go

```diff
@@ -186,42 +202,60 @@ func computeSeedServiceUrl(dc *cassdcapi.CassandraDatacenter) string {
 	return dc.Spec.ClusterName + "-seed-service." + dc.Namespace + ".svc." + clusterDomain
 }
 
-func computeClusterVersion(dc *cassdcapi.CassandraDatacenter) string {
+func computeClusterVersion(dc *cassdcapi.CassandraDatacenter) ClusterVersion {
 	cassandraVersion := dc.Spec.ServerVersion
 	if strings.HasPrefix(cassandraVersion, "3") {
-		return "3.11"
+		return ClusterVersion3
 	} else {
-		return "4.0"
+		return ClusterVersion4
 	}
 }
 
-func computeImage(template *api.StargateTemplate, clusterVersion string) string {
-	containerImage := template.StargateContainerImage
+func computeImage(containerImage *api.ContainerImage, clusterVersion ClusterVersion) string {
 	if containerImage == nil {
-		if clusterVersion == "3.11" {
-			return fmt.Sprintf("%s/%s:v%s", "stargateio", "stargate-3_11", api.DefaultStargateVersion)
+		if clusterVersion == ClusterVersion3 {
+			return DefaultStargate3Image
 		} else {
-			return fmt.Sprintf("%s/%s:v%s", "stargateio", "stargate-4_0", api.DefaultStargateVersion)
+			return DefaultStargate4Image
 		}
 	} else {
 		registry := "docker.io"
-		if containerImage.Registry != nil {
-			registry = *containerImage.Registry
+		if containerImage.Registry != "" {
+			registry = containerImage.Registry
+		}
+		repository := DefaultStargateImageRepository
+		if containerImage.Repository != "" {
+			repository = containerImage.Repository
+		}
+		var name string
+		if containerImage.Name != "" {
+			name = containerImage.Name
+		} else if clusterVersion == ClusterVersion3 {
+			name = DefaultStargate3ImageName
+		} else {
+			name = DefaultStargate4ImageName
 		}
 		tag := "latest"
-		if containerImage.Tag != nil {
-			tag = *containerImage.Tag
+		if containerImage.Tag != "" {
+			tag = containerImage.Tag
 		}
-		return fmt.Sprintf("%v/%v:%v", registry, containerImage.Repository, tag)
+		return fmt.Sprintf("%v/%v/%v:%v", registry, repository, name, tag)
```

Comment:
I believe this was wrong since the beginning, it was concatenating the registry and the repository without an image name.